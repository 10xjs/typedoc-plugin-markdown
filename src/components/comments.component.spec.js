const Handlebars = require('handlebars');
const path = require('path');
const fs = require('fs-extra');

describe(`CommentsComponent`, () => {
  let app;
  let project;
  let pluginInstance;
  const out = path.join(__dirname, 'tmp');
  beforeAll(() => {
    app = bootstrapApp({
      includes: './test/stubs/inc/',
      media: './test/stubs/media/',
      listInvalidSymbolLinks: true,
    });
    project = app.convert(app.expandInputFiles([stubsDirectory]));
    app.generateDocs(project, out);
    pluginInstance = app.renderer.getComponent('comments');
  });

  afterAll(() => {
    fs.removeSync(out);
  });

  test(`should define helper'`, () => {
    const helpers = Handlebars.helpers;
    expect(helpers.comment).toBeDefined();
  });

  test(`should convert symbols brackets to symbol links'`, () => {
    expect(
      Handlebars.helpers.comment.call(project.findReflectionByName('commentsWithSymbolLinks').comment.text),
    ).toMatchSnapshot();
  });

  test(`should set warnings if symbol not found'`, () => {
    expect(pluginInstance.warnings.length > 0).toBeTruthy();
  });

  test(`should convert comments with includes'`, () => {
    expect(
      Handlebars.helpers.comment.call(project.findReflectionByName('commentsWithIncludes').comment.text),
    ).toMatchSnapshot();
  });

  test(`should build @link references'`, () => {
    expect(
      Handlebars.helpers.comment.call(project.findReflectionByName('functionWithDocLink').signatures[0].comment.text),
    ).toMatchSnapshot();
  });
});
