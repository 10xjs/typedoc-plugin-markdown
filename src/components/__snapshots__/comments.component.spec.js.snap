// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsComponent should build @link references' 1`] = `
"Taken from http://usejsdoc.org/tags-inline-link.html.
"
`;

exports[`CommentsComponent should convert comments with includes' 1`] = `
"This is a simple example on how to use include.

![My image alt text](../media/logo.png)

![My not found image](media://VOID.png)
 

This is an example of handlebars include

This is a simple example on a handlebars file.
"
`;

exports[`CommentsComponent should convert symbols brackets to symbol links' 1`] = `
"- Link to an external reflection: [BaseClass](../classes/_classes_.baseclass.md)
- Link to an internal reflection: [commentsInReturn](_comments_.md#commentsinreturn)
- Link to an undefined reflection: [[VOID]]
"
`;
